synapse/storage/databases/main/room.py

```diff
@@ -1285,17 +1286,15 @@ async def is_partial_state_room_batched(
         complete.
         """
 
-        entries = set(
-            await self.db_pool.simple_select_many_batch(
-                table="partial_state_rooms",
-                column="room_id",
-                iterable=room_ids,
-                retcols=("room_id",),
-                desc="is_partial_state_room",
-            )
+        rows: List[Dict[str, Literal[1]]] = await self.db_pool.simple_select_many_batch(
```

Comment:
Isn't this a list of dicts which are each `{"room_id": "!abc:example.com"}`?

Reply:
Oh, you changed the result... hmm....

Reply:
It is. For some reason I thought it was a list of dicts of the form `{"!abc:example.com": "!abc:example.com"}`.

Reply:
```
self.db_pool.simple_select_many_batch(
            table="partial_state_rooms",
            column="room_id",
            iterable=room_ids,
            retcols=("room_id",),
            desc="is_partial_state_room",
        )
```

I thought we were selecting a single column given by the second argument. But no, we are selecting arbitrarily many columns `retcols` WHERE `column` in `iterable`.

On the whole, PICNIC.